lib/at_cmd/at_cmd.c

```diff
@@ -73,44 +73,51 @@ static int open_socket(void)
 	return 0;
 }
 
-static int get_return_code(char *buf, struct resp_item *ret)
+static int get_return_code(char *buf, size_t bytes_read, struct resp_item *ret)
 {
+	bool match;
 	char *tmpstr = NULL;
 	int new_len  = 0;
 
 	ret->state = AT_CMD_NOTIFICATION;
 
 	do {
-		tmpstr = strstr(buf, AT_CMD_OK_STR);
-		if (tmpstr) {
+		/* must match `OK` at the end of the response */
+		tmpstr = buf + bytes_read - ARRAY_SIZE(AT_CMD_OK_STR);
+		match = !strncmp(tmpstr, AT_CMD_OK_STR, strlen(AT_CMD_OK_STR));
+		if (match) {
 			ret->state = AT_CMD_OK;
 			ret->code  = 0;
 			break;
 		}
 
 		tmpstr = strstr(buf, AT_CMD_CMS_STR);
 		if (tmpstr) {
+			match = true;
 			ret->state = AT_CMD_ERROR_CMS;
 			ret->code = atoi(&buf[ARRAY_SIZE(AT_CMD_CMS_STR) - 1]);
 			break;
 		}
 
 		tmpstr = strstr(buf, AT_CMD_CME_STR);
 		if (tmpstr) {
+			match = true;
 			ret->state = AT_CMD_ERROR_CME;
-			ret->code = atoi(&buf[ARRAY_SIZE(AT_CMD_CMS_STR) - 1]);
+			ret->code = atoi(&buf[ARRAY_SIZE(AT_CMD_CME_STR) - 1]);
```

Comment:
There was a typo here, which didn't affect the operation since the two strings have the same length :)